### PR TITLE
MTM-42624 angular 12 upgrade

### PR DESCRIPTION
**Pull request by DawidSAG**

### CODE REVIEW

File: content/web/upgrade-bundle/migration-angular-12.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+title: Upgrading to Angular 12 from 11
+# layout: redirect
+weight: 6
+---
+
+Angular 12 is supported from version `10.11.45.0`. The following configuration changes are required before you can run the application:
+
+- **Enable IVY** (check the section "**How to enable IVY**").
```
Comment (rssag) [inline content/web/upgrade-bundle/migration-angular-12.md:9]:
```suggestion
- Enable IVY, see [How to enable IVY](#how-to-enable-ivy).
```

File: content/web/upgrade-bundle/migration-angular-12.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+- Follow `Angular 12` upgrade guide: [Updating to version 12](https://v12.angular.io/guide/updating-to-version-12).
+- **Use Node** version `14`.
```
Comment (rssag) [inline content/web/upgrade-bundle/migration-angular-12.md:11]:
```suggestion
- Use Node version `14`.
```

File: content/web/upgrade-bundle/migration-angular-12.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+> **Info:** **`AOT`** is not yet supported.
+
+#### How to enable IVY
+
+- edit **tsconfig.json** and change `enableIvy` to `true` or delete it as this setting is the default setting:
```
Comment (rssag) [inline content/web/upgrade-bundle/migration-angular-12.md:17]:
```suggestion
- Edit _tsconfig.json_ and set `enableIvy` to `true` or delete it as this setting is the default setting:
```
where is the tsconfig.json file located?

File: content/web/upgrade-bundle/migration-angular-12.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+```javascript
+    "angularCompilerOptions": {
+        "enableIvy": true
+  }
+```
+
+- edit **package.json** and add `postinstall` hook:
```
Comment (rssag) [inline content/web/upgrade-bundle/migration-angular-12.md:25]:
```suggestion
- Edit _package.json_ and add the `postinstall` hook:
```
where is the package.json file located?

File: content/web/upgrade-bundle/migration-angular-12.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+```json
+    "scripts": {
+        ...
+    "postinstall": "ngcc"
+  }
+```
+- Delete `node_modules` and reinstall them. If `ngcc` works you should see a **compiling** step after installing all dependencies.
+
+- If you use `Jest (JavaScript Testing Framework)` check: [Jest - Angular IVY](https://thymikee.github.io/jest-preset-angular/docs/guides/angular-ivy) for the configuration.
```
Comment (rssag) [inline content/web/upgrade-bundle/migration-angular-12.md:35]:
```suggestion
- If you use Jest (JavaScript Testing Framework) check: [Jest - Angular IVY](https://thymikee.github.io/jest-preset-angular/docs/guides/angular-ivy) for the configuration.
```

File: content/web/upgrade-bundle/migration-angular-12.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+> **Info:** `Deep imports` are not supported. For example, you need to import from **`@c8y/ngx-components/asset-navigator`** but not from the **`@c8y/ngx-components/asset-navigator/services/asset-service`**.
```
Comment (rssag) [inline content/web/upgrade-bundle/migration-angular-12.md:37]:
```suggestion
> **Info:** Deep imports are not supported. For example, you need to import from `@c8y/ngx-components/asset-navigator` but not from `@c8y/ngx-components/asset-navigator/services/asset-service`.
```

File: content/web/upgrade-bundle/migration-angular-12.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+> **Info:** If you are using Visual Studio Code, make sure in the **Angular Language Service Plugin** option: **`"Use legacy View Engine language service"`** is unchecked.
```
Comment (rssag) [inline content/web/upgrade-bundle/migration-angular-12.md:39]:
```suggestion
> **Info:** If you are using Visual Studio Code, make sure that in the Angular Language Service Plugin the option `"Use legacy View Engine language service"` is unchecked.
```

File: content/web/upgrade-bundle/migration-angular-12.md (hunk 3 above)
Comment (DawidSAG) [reply content/web/upgrade-bundle/migration-angular-12.md:17]: > where is the tsconfig.json file located?

It's in the root of the project.

**Review by rssag — DISMISSED** [review]

**Review by janhommes — CHANGES_REQUESTED** [review]

**Review by janhommes — APPROVED** [review]

**Review by BeateRixen — APPROVED** [review]

**Review by MWindel — APPROVED** [review]

**Review by aprimakina — APPROVED** [review]

**Review by BeateRixen — APPROVED** [review]

**Pull request merged**